compiler/ballerina-lang/src/main/java/io/ballerina/projects/repos/FileSystemCache.java

```diff
@@ -98,9 +98,19 @@ public void cacheBir(ModuleName moduleName, ByteArrayOutputStream birContent) {
     public Optional<Path> getPlatformSpecificLibrary(CompilerBackend compilerBackend, String libraryName,
                                                      boolean isOptimizedLibrary) {
         String libraryFileName = libraryName + compilerBackend.libraryFileExtension();
-        Path targetPlatformCacheDirPath = isOptimizedLibrary ? getOptimizedTargetPlatformCacheDirPath(compilerBackend) :
-                getTargetPlatformCacheDirPath(compilerBackend);
-        Path jarFilePath = targetPlatformCacheDirPath.resolve(libraryFileName);
+        Path optimizedJarFilePath = getOptimizedTargetPlatformCacheDirPath(compilerBackend).resolve(libraryFileName);
```

Comment:
We need to get the optimized jar path only if the `isOptimizedLibrary` flag is passed right? We can write the `getOptimizedTargetPlatformCacheDirPath(compilerBackend).resolve(libraryFileName);` part within the if condition without initializing the variable.